### ISSUE TITLE
Audio Message arrange incorrectly

### ISSUE THREAD

**Issue by magicmikek** [jessesquires/JSQMessagesViewController#1891]
I implemented Audio Messages, but unofrtunately they are not displaying correctly? 
Any suggestions to fix this?

![14964100_10209186677002202_2128383792_o](https://cloud.githubusercontent.com/assets/7288679/19988660/c81a71c0-a1f7-11e6-8ad1-08b10234e535.jpg)

**Comment by Lucashuang0802** [comment]
Thanks! 😄

Please provide all the information in the issue template so we can help!
And don't forget to read our contribution guidelines. 😊

**Comment by eliburke** [comment]
@magicmikek search for audio in the closed issues-- i believe there may have been one with a similar issue where either the audio file or the audio control was weakly referenced and went away before it could be rendered on screen.

And as always, reproducing the issue in the demo helps. Make sure your sound file can play there.

**Comment by magicmikek** [comment]
![simulator screen shot 4 nov 2016 15 13 40](https://cloud.githubusercontent.com/assets/7288679/20019152/473b257c-a2a1-11e6-85b5-add5271a1586.png)

It actually works with the audio file from the swift example project. The problem is that I'm using Firebase to store the messages, so I store the object in the database and change the audiourl later, so first it is not set. That probably messes up the audio messages. 
What could be the solution to this?

**Comment by eliburke** [comment]
@magicmikek you won't love my answer 🤕 - you need to extend / customize the audio media item to suit your needs. It was written to be a basic example, easily modified or extended. In this particular case, it assumes the audio file is available. 

I would say you have two options. 1, detect that you don't have the audio file available and return a JSQPhotoMediaItem with a placeholder image (or a button or other custom view of your choice) while it downloads. There is no reason you can't change the type of JSQMediaItem dynamically, just reload afterwards. (this is my particular solution)

Option 2, subclass. Take a look at `JSQAudioMediaItem:mediaView` (it is straightforward and simple). You will see that the first line checks the existence of audioData. If there is none, it returns a nil view. You can create your own MagicMikeAudioMediaItem, override `mediaView`, and check for the existence of the audio data. If there is none, return a UIImageView with a placeholder loaded. If it exists, call `[super mediaView]`. Then when you get the data, call setAudioData or setAudioDataWithUrl to invalidate and redraw.

**Comment by magicmikek** [comment]
I just got a question. I'm doing the same with JSQPhotomediaitem, though there I do not get this error. It just shows me a blank grey JSQPhotoMediaItem with a loading view inside. 
Is it different than the JSQAudioMediaItem by default?

**Comment by eliburke** [comment]
JSQMediaItem has a default method `mediaPlaceholderView` that is inherited by all media subtypes. It will generate placeholder images in cases when mediaView returns a nil value. However, it relies on `mediaViewDisplaySize` to generate the activity indicator. Why don't you debug through and see what is going on there. Maybe it just needs to have a different default size when the audio file is not set.

**Comment by magicmikek** [comment]
So I debugged and found out that it returned a `height = 0` for the `mediaPlaceholderView`. I fixed this and now it returns the default `height = 40` and actually shows a proper loading View for the JSQAudioMediaItem. Unfortunately as soon as I set the audio file and reload the collectionview something odd happens. Check out the pictures:
![simulator screen shot 9 nov 2016 16 32 49](https://cloud.githubusercontent.com/assets/7288679/20143797/59ef4f84-a668-11e6-9283-99d6400692e7.png)
![simulator screen shot 9 nov 2016 16 32 52](https://cloud.githubusercontent.com/assets/7288679/20143809/6410a774-a668-11e6-9bfd-5faf992017f0.png)
![simulator screen shot 9 nov 2016 16 32 54](https://cloud.githubusercontent.com/assets/7288679/20143810/6660b280-a668-11e6-94bf-2ee4f084f5b7.png)
